runtime/runtime/src/lib.rs

```diff
@@ -888,6 +890,13 @@ impl Runtime {
             }
         }
 
+        for account_id in validator_accounts_update.slashed_accounts.iter() {
+            if let Some(mut account) = get_account(state_update, &account_id)? {
+                account.locked = 0;
+                set_account(state_update, &account_id, &account);
```

Comment:
1. Should the locked amount be transferred to the treasury?
2. I assume somewhere else we are immediately kicking out this validator and not waiting for it to be automatically kicked out at the end of the epoch. Correct?

Reply:
@ilblackdragon  Is (1) resolved? What about (2)?

Reply:
@ilblackdragon ^

Reply:
Yes, slashing marks the account in epoch_manager as slashed and the blocks from it are considered invalid going forward. It gets removed from the list after end of the epoch.